### ISSUE TITLE
How can I install CPU_only version from source?

### ISSUE THREAD

**Issue by yd-yin** [facebookresearch/pytorch3d#1233]
I always get CUDA issues when trying to install GPU supported PyTorch3D.

If I only want CPU_only PyTorch3D, how can I achieve that?

**Comment by yd-yin** [comment]
I'm commenting to remove stale label

**Comment by bottler** [comment]
The easiest way in released versions might be to edit setup.py to not build the cuda extension (the equivalent of `torch.cuda.is_available()` being False). 

On linux, I can get the same effect by setting CUDA_VISIBLE_DEVICES to the empty string, but this may be fragile. (e.g. `CUDA_VISIBLE_DEVICES= python -c "import torch; print(torch.cuda.is_available())"` prints `False`).

After a fix we just made on the main branch, you can just set `PYTORCH3D_FORCE_NO_CUDA=1` in the environment to get what you want cleanly.

**Comment by ecnuycxie** [comment]
build from source
